csrc/multidevice/communication.cpp

```diff
@@ -456,11 +453,16 @@ c10::intrusive_ptr<c10d::Work> postReduceScatter(
     c10d::Backend* backend,
     at::Tensor input_tensor,
     at::Tensor output_tensor) {
-  const auto scattered_axis = communication->scatteredAxis();
-  NVF_ERROR(
-      scattered_axis >= 0,
-      "scattered_axis is expected to be non-negative: ",
-      scattered_axis);
+  TensorView* output_tv = communication->out();
+  auto reduction_axis = output_tv->getReductionAxis().value();
+  auto scattered_axis = getShardedLogicalAxis(output_tv, ParallelType::DIDx);
```

Comment:
Can we check DIDx is the only DID type in output_tv? Otherwise, things will quietly fail when we extend ReduceScatter to 2D sharding, making debugging harder.